### ISSUE TITLE
Feature Request: Option to write live text into a text file

### ISSUE THREAD

**Issue by nyszilard** [quelea-projection/Quelea#313]
With this option, it would be possible to use lyrics in OBS as subtitle without any workaround.

**Expected options on the UI**
[x] Write live text into file
Path: |input field| Browse...

**Expected behavior**
- onBlackScreen, onClearScreen, onExit it should clear the text file
- onGoLive, onSwitch it should replace the content of the text file with the selected part of the lyrics

**Current workaround**
I am currently using the mobile lyrics feature to extract the text with a nodejs script.

**Comment by ArvidNy** [comment]
I'm sorry but I'm not sure I fully understand what you mean. I haven't used OBS myself so I'm not sure how text files can be used. All I know is that users have successfully used the projection screen with an invisible background to add the lyrics as subtitles to the stream.

What would be the advantages of exporting the lyrics to a text file?

**Comment by berry120** [comment]
If I've understood this correctly I can see the value in it - if Quelea updates a text file "live" with the lyrics its showing, then you have the possibility in OBS to read that text in and format it, style it, etc. over a background, video feed etc. of your choice.

If that's the case then I think this could well be worthwhile. It would have to come with the caveat though that this would only work for songs & bible passages - obviously we couldn't do anything with images, powerpoint presentations, videos etc.

**Comment by PaulD987** [comment]
What would be done with the bible verse numbers? Would these be left in the text? or removed completely. Obviously would lose the formatting when writing out to the file. So, just just leave them in (but remove the internal <sup></sup> formatting tags), or remove the verse numbers before writing out to the file. Seems a relatively simple addition to make.

**Comment by kavishdahekar** [comment]
This is an interesting idea! However, I'm wondering how much latency the file r/w operations would add both to quelea as well as any other application that wants to read the text file.

For a quick test, I added a Text (FreeType 2) source to OBS with the option to read from a text file. Any modifications to the text file do reflect in OBS but with a significant delay(almost 0.5 to 1 second). I have a decently powerful system, so I don't think it's a slow h/w issue.

For our church's use-case, I have tried directly adding the mobile lyrics server as browser source in OBS but it comes with a #cccccc coloured background that I then need to remove. I have also tried directly adding the mobile lyrics server's`/lyrics` endpoint as a browser source, but it outputs the lyrics in a `<span class="lyric">` tag which OBS does not recognise as valid html and displays it all as plain text.

I suggested in #346 that we have an api that would make such integrations easier, but understandably thats a high effort task with low returns.
Maybe we could atleast allow the mobile lyrics server's `/lyrics` response format to be controlled via a query param.
`/lyrics?o=txt` will respond with plain text only (we'll have to figure out how to deal with bible chapter/verse number).
we could respond with text,json,html to allow various ways to consume. Omitting the `o` query param would default to current behavior of returning the `span`.

Would that help your use-case @nyszilard ?
